UnitySetup/UnitySetup.psm1

```diff
@@ -2254,8 +2246,18 @@ Would you like to continue? (Default: $($createPAT))"
             }
             Connect-AzAccount @connectArgs  | Out-Null
         }
-        $AZTokenRequest = Get-AzAccessToken -ResourceType Arm
-        $headers = @{ Authorization = "Bearer $($AZTokenRequest.Token)" }
+
+        $AZToken = $null
+        $azAccountsVersion = Get-ModuleVersion -ModuleName "Az.Accounts"
```

Comment:
The `-AsSecureString` parameter has been there since [2.17](https://github.com/Azure/azure-powershell/blob/fde504cfde96e797787698a40e1692a2d4b66a4a/documentation/SyntaxChangeLog/SyntaxChangeLog-Az11.md?plain=1#L54) release in April 2024 - the recommendation seems to switch to always using it, which I think would avoid this branch and the need to check the version.

Perhaps we require at least this version in the PSD1 and we also use the param and avoid the branch?

Reply:
Sure I think requiring 2.17 minimum and removing the `if` is a fine approach.  Yeah the secure string flag was what the library's output suggested we use to 'void' the change - though it did require a minor change here anyway.

![image](https://github.com/user-attachments/assets/6571949f-aba0-4764-944e-9f61d0e48499)